io/io/inc/TFileMergeServer.h

```diff
@@ -0,0 +1,85 @@
+#ifndef ROOT_TFileMergeServer
+#define ROOT_TFileMergeServer
+
+#include "TBits.h"
+#include "TFile.h"
+#include "TFileCacheWrite.h"
+#include "TFileMerger.h"
+#include "TTimeStamp.h"
+
+#include <condition_variable>
+#include <exception>
+#include <stdexcept>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <thread>
+
+namespace ROOT {
+namespace Experimental {
+
+class TFileMergeClient;
+
+class TFileMergeServer {
+   public:
+      TFileMergeServer() = delete;
+      TFileMergeServer(const TFileMergeServer&) = delete;
+
+      TFileMergeServer(const std::string& name) : fName(name)
+      {
+         fMergingThread.reset(new std::thread([this]() { this->Listen(); }));
```

Comment:
This comment is vague. Please suggest a better name for the variable and explain what you mean by outlining the constructor.